### PR TITLE
Initial PLD implementation

### PR DESCRIPTION
**Pull request by bpiddubnyi**
Ported the precision landing/IR-LOCK system into Quadplane.
This commit based on the
#19377
With the rebase over latest master

### CODE REVIEW

File: libraries/AP_Scheduler/AP_Scheduler.cpp [hunk 1]
```diff
@@ -34,7 +34,7 @@
 #endif
 #include <stdio.h>
 
-#if APM_BUILD_COPTER_OR_HELI || APM_BUILD_TYPE(APM_BUILD_ArduSub)
+#if APM_BUILD_COPTER_OR_HELI || APM_BUILD_TYPE(APM_BUILD_ArduSub) || APM_BUILD_TYPE(APM_BUILD_ArduPlane)
```
Comment (amilcarlucas) [inline libraries/AP_Scheduler/AP_Scheduler.cpp:37]: This will for sure have some side effects.

**Review by amilcarlucas — COMMENTED** [review]
I wonder how much of this can actually be moved to AP_vehicle so that other vehicles, like the blimp can re-use it as well.

**Comment by peterbarker** [comment]
We've just done this for Rover in https://github.com/ArduPilot/ardupilot/pull/21132

As soon as that's merged I was planning on factoring it all up into AP_Vehicle.  I haven't requested our GSoC student make that change himself given his limited time to work on his project.

Also - needs autotests.

**Comment by fisian** [comment]
I tried this out on a QuadPlane with an IR-Lock sensor.
The target detection worked as expected, however the plane did not use it for corrections in QLoiter/QLand.
I will check what might be wrong.

Is this the latest PR/rebase of precision landing?
I'm open to help testing this feature.

**Comment by tridge** [comment]
i've decided to implement this in scripting for flexibility: https://github.com/ArduPilot/ardupilot/pull/26380

**Pull request closed without merging**